edb/language_server/definition.py

```diff
@@ -114,18 +114,83 @@ def _get_definition_in_ql(
         ls.show_message_log(f"don't know how to lookup schema by {node}")
         return None
 
-    span: edb_span.Span | None = target.get_span(schema)
+    return _schema_obj_to_doc_location(ls, target, document)
+
+
+# Finds definition of names in schema files.
+#
+# Parses the file and finds the ObjectRef at the given position. Then, it
+# computes "module context", by looking at names of encapsulating modules so
+# it can convert ObjectRef into a qualified name. Then it just looks up that
+# name in the schema.
+#
+# This impl might be lacking, since it does not use the code we use for name
+# resolution in the main compiler (tracing.py), and might report some
+# definitions incorrectly (i.e. within expressions).
+def _get_definition_in_schema(
+    ls: ls_server.GelLanguageServer,
+    document: pygls.workspace.TextDocument,
+    position: int,
+) -> lsp_types.Location | None:
+    res = ls_schema._ensure_schema_docs_loaded(ls)
+    if res.err:
+        return None
+
+    # parse current doc, return on errors
+    _ = ls_schema._parse_schema(ls)
+    assert ls.state.schema_sdl
+
+    # find the span in ql ast
+    node_path = edb_span.find_by_source_position(ls.state.schema_sdl, position)
+    if not node_path:
+        return None
+
+    # ls.show_message_log(f"found node: {dump_to_str(node_path[0])}")
+
+    # only resolve ObjectRefs
+    if not isinstance(node_path[0], qlast.ObjectRef):
+        return None
```

Comment:
Make sure pointers are on a TODO list somewhere

Reply:
I guess more broadly I mean, "understand expressions nested in the schema" better?

Reply:
I don't think we have any place in schema where we would allow pointer references that is not a `s_expr.Expression`.

And "understand expressions nested in the schema" is on the list, yes